src/priceHistory/priceHistoryChart.tsx

```diff
@@ -0,0 +1,357 @@
+  const onLayout = useCallback(
+    ({
+      nativeEvent: {
+        layout: { width },
+      },
+    }: LayoutChangeEvent) => {
+      if (Math.abs(width - chartWidth) > 2) {
+        if (x - width / 2 - horizontalOffset < 0) {
+          setAdjustedX(width / 2 + horizontalOffset)
+        }
+        if (x + width / 2 + horizontalOffset > chartWidth) {
+          setAdjustedX(chartWidth - width / 2 - horizontalOffset)
+        }
+      }
+    },
+    [x]
+  )
+  return (
+    <SvgText
+      /*
+      // @ts-ignore */
+      onLayout={onLayout}
+      fill={colors.gray4}
+      fontSize="14"
+      fontFamily="Inter-Regular"
+      x={adjustedX}
+      y={y}
+      textAnchor="middle"
+    >
+      {value}
+    </SvgText>
+  )
+}
+
+function renderPointOnChart(
+  chartData: Array<{ amount: number | BigNumber; displayValue: string }>,
+  chartWidth: number,
+  color: colors
+) {
+  let lowestRateIdx = 0,
+    highestRateIdx = 0
+  chartData.forEach((rate, idx) => {
+    if (rate.amount > chartData[highestRateIdx].amount) {
+      highestRateIdx = idx
+    }
+    if (rate.amount < chartData[lowestRateIdx].amount) {
+      lowestRateIdx = idx
+    }
+  })
+  return ({ datum, x, y }: { x: number; y: number; datum: { _x: number; _y: number } }) => {
+    const idx = datum._x
+    const result = []
+    switch (idx) {
+      case 0:
+        result.push(
+          <G key={idx + 'dot'}>
+            <Line x1={0} y1={y} x2={chartWidth} y2={y} stroke={colors.gray2} strokeWidth="1" />
+            <Circle cx={x} cy={y} r="4" fill={color} />
+          </G>
+        )
+        break
+
+      case chartData.length - 1:
+        result.push(
+          <G key={idx + 'dot'}>
+            <Circle cx={x} cy={y} r="4" fill={color} />
+          </G>
+        )
+        break
+    }
+    switch (idx) {
+      case highestRateIdx:
+        result.push(
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={idx}
+            value={chartData[highestRateIdx].displayValue}
+            position={'top'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+
+      case lowestRateIdx:
+        result.push(
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={idx}
+            value={chartData[lowestRateIdx].displayValue}
+            position={'bottom'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+    }
+
+    switch (result.length) {
+      case 0:
+        return null
+      case 1:
+        return result[0]
+      default:
+        return <>{result}</>
+    }
+  }
+}
+
+interface PriceHistoryChartProps {
+  tokenId: string
+  containerStyle?: ViewStyle
+  testID?: string
+  chartPadding?: number
+  color?: colors
+}
+
+export default function PriceHistoryChart({
+  tokenId,
+  containerStyle,
+  testID,
+  chartPadding,
+  color = colors.black,
+}: PriceHistoryChartProps) {
+  const priceHistoryPrices = useSelector((state: RootState) =>
```

Comment:
nit: `historicalPrices` over `priceHistoryPrices`? same for the selector

Reply:
Ended up going with `prices` and `status` in b52551b3dd85ab7fd661e85bd76ecf3eba8c5b0e.